catalog_tools/analysis/estimate_beta.py

```diff
@@ -230,6 +233,167 @@ def estimate_b_laplace(
+    complete_events = pd.DataFrame.groupby(
+        pd.DataFrame(data={
+            'mag_left_edge': np.array(
+                [i.left for i in pd.cut(magnitudes[idxcomp],
+                 bins=np.arange(completeness_magnitudes[0],
+                 mag_max + 0.01, delta_m), right=False)]),
+            'completeness_start': cmpl_yrs[idxcomp]
+        }), by=['mag_left_edge', 'completeness_start'])\
+        .size()\
+        .to_frame('num')\
+        .reset_index()
+    assert np.all(
+        complete_events.completeness_start > 0)  # should be the case by design
+
+    # minimization
+    beta = np.log(10)  # initialization of beta
+    output = minimize(weichert_minimization,
+                      beta,
+                      args=(tend, complete_events, delta_m),
+                      method='Nelder-Mead',
+                      options={'maxiter': 5000, 'disp': True},
+                      tol=1e5 * np.finfo(float).eps)
+    b_parameter = output.x[0] * factor
+
+    # compute rate at lower magnitude of completeness bin
+    wf = np.sum(np.exp(-output.x[0]
+                       * (complete_events.mag_left_edge + delta_m * 0.5)))\
+        / np.sum((tend - complete_events.completeness_start.values)
+                 * np.exp(-output.x[0]
+                          * (complete_events.mag_left_edge + delta_m * 0.5)))
+    rate_at_lmc = complete_events.num.sum() * wf
+
+    # compute a-value ( a_val = log10(rate at M=0) )
+    a_val = np.log10(rate_at_lmc) + \
+        (output.x[0] / np.log(10)) * complete_events.mag_left_edge.values[0]
+
+    # compute uncertainty in b-parameter according to Weichert (1980)
+    nominator = np.sum((tend - complete_events.completeness_start.values)
+                       * np.exp(-output.x[0] * (complete_events.mag_left_edge
+                                                + delta_m * 0.5)))**2
+    denominator_term1 = np.sum(
+        (tend - complete_events.completeness_start.values)
+        * (complete_events.mag_left_edge + delta_m * 0.5)
+        * np.exp(- output.x[0] * (complete_events.mag_left_edge
+                 + delta_m * 0.5)))**2
+    denominator_term2 = np.sqrt(nominator) * \
+        np.sum((tend - complete_events.completeness_start.values)
+               * ((complete_events.mag_left_edge + delta_m * 0.5)**2)
+               * np.exp(-output.x[0] * (complete_events.mag_left_edge
+                        + delta_m * 0.5))
+               )
+    var_beta = -(1 / complete_events.num.sum()) * nominator\
+        / (denominator_term1 - denominator_term2)
+    std_b_parameter = np.sqrt(var_beta) * factor
+
+    # compute uncertainty in rate at lower magnitude of completeness
+    std_rate_at_lmc = rate_at_lmc / np.sqrt(complete_events.num.sum())
+
+    return b_parameter, std_b_parameter, rate_at_lmc, std_rate_at_lmc, a_val
+
+
+def weichert_minimization(beta, tend, complete_events, delta_m):
+    magbins = complete_events.mag_left_edge + delta_m * 0.5
```

Comment:
It is an internal function that needed to be defined for passing it to scipy.optimize.minimize. I renamed it to _weichert_objective_function with the leading underscore to indicate that is just used internally. Added a short docstring too. 

Alternatively, maybe:
1) it can be defined as a nested function within estimate_b_weichert? I don't like it because it is a bit too complex and I will need to rename the arguments and it will get confusing.
2) add a new utils submodule (catalog_tools.analysis.utils) and import it from there. Other helper functions (e.g. the function 'differences' you also have in estimate_beta.py) can go there too.